src/backend/Csrs.Api/Repositories/Repository.cs

```diff
@@ -1,14 +1,89 @@
-﻿using Simple.OData.Client;
+﻿using Csrs.Api.Models.Dynamics;
+using Simple.OData.Client;
 
 namespace Csrs.Api.Repositories
 {
-    public abstract class Repository
+    /// <summary>
+    /// Implements the generic repository interface
+    /// </summary>
+    /// <typeparam name="TEntity">The type of entity.</typeparam>
+    public abstract class Repository<TEntity> : IRepository<TEntity> where TEntity : Entity
     {
+        /// <summary>
+        /// The <see cref="IODataClient"/>.
+        /// </summary>
         protected readonly IODataClient Client;
         
         protected Repository(IODataClient client)
         {
             Client = client ?? throw new ArgumentNullException(nameof(client));
         }
+
+        public async Task DeleteAsync(TEntity entity, CancellationToken cancellationToken)
+        {
+            if (entity is null)
+            {
+                throw new ArgumentNullException(nameof(entity));
+            }
+
+            await Client
+                .For<TEntity>()
+                .Key(entity.Id)
+                .DeleteEntryAsync(cancellationToken);
+        }
+
+        public async Task<TEntity?> GetAsync(Guid id, System.Linq.Expressions.Expression<Func<TEntity, object>> properties, CancellationToken cancellationToken)
+        {
+            if (properties is null)
+            {
+                throw new ArgumentNullException(nameof(properties));
+            }
+
+            try
+            {
+                TEntity entity = await Client
+                    .For<TEntity>()
+                    .Key(id)
+                    .Select(properties)
+                    .FindEntryAsync(cancellationToken);
+
+                return entity;
+            }
+            catch (HttpRequestException exception) when (exception.IsTimedOut())
+            {
+                throw;
```

Comment:
This is just an example of the type of exception that will be thrown when not connected. In future changes, we will add custom exception types, logging and incrementing metrics indicating that connection time out occured.